editor/modes/visual-editor/block-list.js

```diff
@@ -56,8 +56,10 @@ class VisualEditorBlockList extends wp.element.Component {
 		this.setState( { selectionAtStart: null } );
 	}
 
-	handleClickOutside() {
-		this.props.clearSelectedBlock();
+	handleClickOutside( { target } ) {
+		if ( target.nodeName !== 'BUTTON' ) {
```

Comment:
Is this specifically targeting the `<MultiSelectHeader />` button? Are we trying to resolve a race condition between when deleting blocks from header, and clicking outside? Is there any other way we could ensure the "Delete" logic is triggered first? `setTimeout` is an option; not great, but perhaps still better than targeting other elements on the page.